backend/infrahub/menu/models.py

```diff
@@ -1,9 +1,10 @@
 from __future__ import annotations
 
 from dataclasses import dataclass, field
-from typing import TYPE_CHECKING, Self
+from typing import TYPE_CHECKING
 
 from pydantic import BaseModel, Field
+from typing_extensions import Self
```

Comment:
Are these changes something that happened when the wrong Python version was running? It doesn't seem like something that would be required.

Reply:
Yes and No
it was uncovered because the tests did run with 3.10 instead of 3.12 but officially we should support 3.10 (pyproject.toml) so we shouldn't use a 3.11 feature.